packages/devtools/lib/src/timeline/event_details.dart

```diff
@@ -100,43 +102,33 @@ class EventDetails extends CoreElement {
       'CPU Flame Chart (preview)',
       EventDetailsTabType.flameChart,
     );
-    final bottomUpTab = EventDetailsTabNavTab(
-      'Bottom Up',
-      EventDetailsTabType.bottomUp,
-    );
     final callTreeTab = EventDetailsTabNavTab(
       'Call Tree',
       EventDetailsTabType.callTree,
     );
+    final bottomUpTab = EventDetailsTabNavTab(
+      'Bottom Up',
+      EventDetailsTabType.bottomUp,
+    );
 
     tabNav = PTabNav(<EventDetailsTabNavTab>[
       flameChartTab,
-      bottomUpTab,
       callTreeTab,
+      bottomUpTab,
     ])
       ..element.style.borderBottom = '0';
 
     _selectedTab = flameChartTab;
 
-    tabNav.onTabSelected.listen((PTabNavTab tab) async {
+    tabNav.onTabSelected.listen((PTabNavTab tab) {
       // Return early if this tab is already selected.
       if (tab == _selectedTab) {
         return;
       }
       _selectedTab = tab;
 
       assert(tab is EventDetailsTabNavTab);
-      switch ((tab as EventDetailsTabNavTab).type) {
-        case EventDetailsTabType.flameChart:
-          await uiEventDetails.showTab(EventDetailsTabType.flameChart);
-          break;
-        case EventDetailsTabType.bottomUp:
-          await uiEventDetails.showTab(EventDetailsTabType.bottomUp);
-          break;
-        case EventDetailsTabType.callTree:
-          await uiEventDetails.showTab(EventDetailsTabType.callTree);
-          break;
-      }
+      uiEventDetails.showTab((tab as EventDetailsTabNavTab).type);
```

Comment:
nit: having both the is and as checks isn't needed.

Reply:
removed the assert